src/erlcloud_ddb2.erl

```diff
@@ -1871,6 +2027,69 @@ get_item(Table, Key, Opts, Config) ->
     out(Return, fun(Json, UOpts) -> undynamize_record(get_item_record(), Json, UOpts) end, DdbOpts, 
         #ddb2_get_item.item, {ok, []}).
 
+%%%------------------------------------------------------------------------------
+%%% ListGlobalTables
+%%%------------------------------------------------------------------------------
+
+-type list_global_tables_opt() :: {limit, pos_integer()} | 
+                                  {exclusive_start_global_table_name, table_name() | undefined} |
+                                  out_opt().
+-type list_global_tables_opts() :: [list_global_tables_opt()].
+
+-spec list_global_tables_opts() -> opt_table().
+list_global_tables_opts() ->
+    [{limit, <<"Limit">>, fun id/1},
+     {exclusive_start_global_table_name, <<"ExclusiveStartGlobalTableName">>, fun id/1}].
+
+-spec undynamize_global_tables(jsx:json_term(), undynamize_opts()) -> [#ddb2_global_table{}].
+undynamize_global_tables(Tables, Opts) ->
+    [undynamize_record(global_table_record(), T, Opts) || T <- Tables].
+
+-spec list_global_tables_record() -> record_desc().
+list_global_tables_record() ->
+    {#ddb2_list_global_tables{},
+     [{<<"GlobalTables">>, #ddb2_list_global_tables.global_tables, fun undynamize_global_tables/2},
+      {<<"LastEvaluatedGlobalTableName">>, #ddb2_list_global_tables.last_evaluated_global_table_name, fun id/2}
+     ]}.
+
+-type list_global_tables_return() :: ddb_return(#ddb2_list_global_tables{}, [#ddb2_global_table{}]).
+
+-spec list_global_tables() -> list_global_tables_return().
+list_global_tables() ->
+    list_global_tables([], default_config()).
+
+-spec list_global_tables(list_global_tables_opts()) -> list_global_tables_return().
+list_global_tables(Opts) ->
```

Comment:
add a pedantic guard for `when is_list(Opts)`
as other apis allow `/1 (#aws_config{})`